controllers/operatorpolicy_controller.go

```diff
@@ -60,166 +65,201 @@ var _ reconcile.Reconciler = &OperatorPolicyReconciler{}
 func (r *OperatorPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	policy := &policyv1beta1.OperatorPolicy{}
 
-	err := r.Get(ctx, req.NamespacedName, policy)
+	watcher := depclient.ObjectIdentifier{
+		Group:     operatorv1.GroupVersion.Group,
+		Version:   operatorv1.GroupVersion.Version,
+		Kind:      "OperatorPolicy",
+		Namespace: req.Namespace,
+		Name:      req.Name,
+	}
+
+	// Start query batch for caching and watching related objects
+	err := r.DynamicWatcher.StartQueryBatch(watcher)
```

Comment:
It might be good to `defer` stopping the query batch, but that's a change we can consider in a future PR.

Reply:
As mentioned in Slack, this wasn't addressed. You either need to move this after retrieving the operator policy or move the defer up since it looks like in the unknown error case, you currently aren't ending the query batch which will cause an issue on the next reconcile.